src/commands/profiles/delete.ts

```diff
@@ -0,0 +1,354 @@
+      { name: "@fiscalCode", value: fiscalCode }
+    ];
+
+    const selectFromRecipientFiscalCode =
+      "SELECT * FROM c WHERE c.recipientFiscalCode = @recipientFiscalCode";
+    const paramsFromRecipientFiscalCode: cosmos.SqlParameter[] = [
+      { name: "@recipientFiscalCode", value: fiscalCode }
+    ];
+
+    // retrive local azure credianial
+    cli.action.start("Retrieving cosmosdb credentials");
+    const [connection, storageConnection] = await Promise.all([
+      getCosmosConnection("agid-rg-test", "agid-cosmosdb-test"),
+      getStorageConnection(config.storageName)
+    ]);
+    cli.action.stop();
+
+    // define all delete operations
+    const deleteOps: ReadonlyArray<DeleteOp> = [
+      {
+        containerName: "profiles",
+        query: selectFromFiscalCode,
+        queryParameters: paramsFromFiscalCode,
+        relatedOps: [],
+        deleteBlobs: false
+      },
+      {
+        containerName: "messages",
+        query: selectFromFiscalCode,
+        queryParameters: paramsFromFiscalCode,
+        relatedOps: [
+          {
+            containerName: "message-status"
+          }
+        ],
+        deleteBlobs: true
+      },
+      {
+        containerName: "notifications",
+        query: selectFromFiscalCode,
+        queryParameters: paramsFromFiscalCode,
+        queryOptions: { enableCrossPartitionQuery: true },
+        relatedOps: [
+          {
+            containerName: "notification-status"
+          }
+        ],
+        deleteBlobs: false
+      },
+      {
+        containerName: "sender-services",
+        query: selectFromRecipientFiscalCode,
+        queryParameters: paramsFromRecipientFiscalCode,
+        relatedOps: [],
+        deleteBlobs: false
+      }
+    ];
+
+    // define the delete operation set from given inputs
+    // if all flag is enabled all above defined options will be procecess
+    // otherwhise only operations specified in the given flags
+    const deleteOpsToProcess = flags.all
+      ? deleteOps
+      : deleteOps.filter(x => {
+          switch (x.containerName) {
+            case "messages":
+            case "message-status":
+              return flags.message;
+            case "notifications":
+            case "notification-status":
+              return flags.notification;
+            case "profiles":
+              return flags.profile;
+            case "sender-services":
+              return flags.service;
+          }
+        });
+    if (deleteOpsToProcess.length === 0) {
+      cli.error("please specify at least 1 container");
+    }
+
+    const client = new cosmos.CosmosClient({
+      endpoint: connection.endpoint,
+      auth: { key: connection.key }
+    });
+    const database = client.database("agid-documentdb-test");
+
+    // tslint:disable-next-line: no-let
+    let countDeleteItems = 0;
+    const deleteOperations = (ops: ReadonlyArray<DeleteOp>) => {
+      return ops.reduce((p, file) => {
```

Comment:
extract the function that runs the promises sequentially to a helper, it's not immediately clear what this code does